File: Examples/c/02_variables_types.c
/*
 * Variables and Data Types
 * 
 * Demonstrates:
 * - Different data types (int, float, char, string)
 * - Variable declaration and initialization
 * - Format specifiers in printf()
 */

#include <stdio.h>

int main() {
    // Integer variables
    int age = 25;
    int year = 2025;
    
    // Floating point
    float price = 19.99;
    float temperature = 72.5;
    
    // Characters and strings
    char grade = 'A';
    char name[50] = "Alice";
    
    // Display all variables
    printf("=== Variable Examples ===\n\n");
    
    printf("Name: %s\n", name);
    printf("Age: %d years\n", age);
    printf("Grade: %c\n", grade);
    printf("Year: %d\n", year);
    printf("Price: $%.2f\n", price);
    printf("Temperature: %.1f°F\n", temperature);
    
    // Arithmetic with variables
    int next_year = year + 1;
    float total = price * 2;
    
    printf("\nCalculations:\n");
    printf("Next year: %d\n", next_year);
    printf("Two items: $%.2f\n", total);
    
    return 0;
}

File: examples/pascal/02_variables_types.pas
{
  Variables and Data Types
  
  Demonstrates:
  - Variable declarations (Var section)
  - Different types: Integer, Real, Char, String
  - Assignment with :=
  - Formatted output
}

Program VariablesAndTypes;
Var
  age: Integer;
  price: Real;
  grade: Char;
  name: String;
  year: Integer;
  temperature: Real;

Begin
  { Initialize variables }
  name := 'Alice';
  age := 25;
  grade := 'A';
  year := 2025;
  price := 19.99;
  temperature := 72.5;
  
  WriteLn('=== Variable Examples ===');
  WriteLn;
  
  WriteLn('Name: ', name);
  WriteLn('Age: ', age, ' years');
  WriteLn('Grade: ', grade);
  WriteLn('Year: ', year);
  WriteLn('Price: $', price:0:2);
  WriteLn('Temperature: ', temperature:0:1, '°F');
  
  { Arithmetic operations }
  WriteLn;
  WriteLn('Calculations:');
  WriteLn('Next year: ', year + 1);
  WriteLn('Two items: $', (price * 2):0:2);
End.
